src/localization/gps_linearization.py

```diff
@@ -28,59 +31,90 @@ class GPSLinearization:
     ref_lon: float
     ref_alt: float
 
-    # frame configuration
+    # frame/publishing configuration
+    publish_tf: bool
     use_odom: bool
     world_frame: str
     odom_frame: str
     rover_frame: str
 
-    def __init__(self):
-        # init to zero pose
-        self.pose = SE3()
-
-        rospy.Subscriber("gps/fix", NavSatFix, self.gps_callback)
-        rospy.Subscriber("imu/data", ImuAndMag, self.imu_callback)
-
-        self.tf_broadcaster = tf2_ros.TransformBroadcaster()
-        self.tf_buffer = tf2_ros.Buffer()
-        self.tf_listener = tf2_ros.TransformListener(self.tf_buffer)
+    # covariance config
+    use_dop_cov: bool
+    config_gps_covariance: np.ndarray
 
+    def __init__(self):
         # read required parameters, if they don't exist an error will be thrown
         self.ref_lat = rospy.get_param("gps_linearization/reference_point_latitude")
         self.ref_lon = rospy.get_param("gps_linearization/reference_point_longitude")
         self.ref_alt = rospy.get_param("gps_linearization/reference_point_altitude")
 
-        self.use_odom = rospy.get_param("gps_linearization/use_odom_frame")
-        self.world_frame = rospy.get_param("gps_linearization/world_frame")
-        self.odom_frame = rospy.get_param("gps_linearization/odom_frame")
-        self.rover_frame = rospy.get_param("gps_linearization/rover_frame")
+        self.publish_tf = rospy.get_param("gps_linearization/publish_tf")
```

Comment:
so if I understand correctly, what happens here is that if publish_tf is on, the output goes directly to the tf tree, otherwise the output goes to the EKF which then outputs to the TF tree? If so, this needs to have a comment explaining that. 

 Also, EKF is only relevant for the map -> rover stuff right, no effect on odom-> rover?

Reply:
thats correct

Reply:
yeah I think thats a good idea